src/components/v2/OnlinePreviewPanel.vue

```diff
@@ -0,0 +1,195 @@
+});
+
+function getReadableDate(date: string): string {
+    return valueToReadableDate(new Date(date));
+}
+
+function getReadableCategories(mod: ThunderstoreMod) {
+    return mod.getCategories().join(", ") || "None";
+}
+
+const markdownToRender = computed(() => {
+    switch (activeTab.value) {
+        case "README": return readme.value;
+        case 'CHANGELOG': return changelog.value;
+        default: return null;
+    }
+});
+
+function showDownloadModal(mod: ThunderstoreMod) {
+    store.commit("openDownloadModModal", mod);
+}
+
+</script>
+
+<template>
+    <div class="c-preview-panel">
+        <div class="c-preview-panel__header">
+            <h1 class="title">{{ mod.getName() }}</h1>
+            <h2 class="subtitle">
+                By {{ mod.getOwner() }}
+            </h2>
+            <p class='card-timestamp'><strong>Downloads:</strong> {{mod.getDownloadCount()}}</p>
+            <p class='card-timestamp'><strong>Likes:</strong> {{mod.getRating()}}</p>
+            <p class='card-timestamp'><strong>Last updated:</strong> {{getReadableDate(mod.getDateUpdated())}}</p>
+            <p class='card-timestamp'><strong>Categories:</strong> {{getReadableCategories(mod)}}</p>
+            <div class="margin-top">
+                <p class="description">{{ mod.getDescription() }}</p>
+            </div>
+        </div>
+        <div class="sticky-top inherit-background-colour sticky-top--no-shadow sticky-top--opaque no-margin sticky-top--no-padding">
+            <div class="pad pad--top">
+                <button class="button is-info margin-right" @click="showDownloadModal(mod)">Download</button>
+                <Link target="_blank" tag="button" class="button" :url="props.mod.getPackageUrl()">View online</Link>
+            </div>
+            <div class="tabs margin-top">
+                <ul>
+                    <li :class="{'is-active': activeTab === 'README'}"><a @click="setActiveTab('README')">README</a></li>
+                    <li :class="{'is-active': activeTab === 'CHANGELOG'}"><a @click="setActiveTab('CHANGELOG')">CHANGELOG</a></li>
+                    <li :class="{'is-active': activeTab === 'Dependencies'}"><a @click="setActiveTab('Dependencies')">Dependencies ({{ dependencies.length }})</a></li>
+                </ul>
+            </div>
+        </div>
+        <div class="c-preview-panel__content">
+            <template v-if="loadingPanel">
+                <div class="notification">
+                    <div class="container">
+                        <p>Fetching {{ activeTab }} for {{ props.mod.getFullName() }}</p>
+                    </div>
+                </div>
+            </template>
+            <template v-else-if="activeTab === 'Dependencies'">
+                <template v-if="dependencies.length > 0">
+                    <OnlineModList :paged-mod-list="dependencies" :read-only="true" />
+                </template>
+                <template v-else>
+                    <p>No dependencies</p>
```

Comment:
This could use the "notification"/"container" wrappers to visually match "No changelog" case.